src/gui/TotpSetupDialog.cpp

```diff
@@ -43,51 +45,92 @@ TotpSetupDialog::~TotpSetupDialog()
 
 void TotpSetupDialog::saveSettings()
 {
+    // Secret key sanity check
+    auto key = m_ui->seedEdit->text().toLatin1();
+    auto sanitizedKey = Base32::sanitizeInput(key);
+    if (sanitizedKey != key) {
+        MessageBox::information(this,
+                                "Invalid TOTP Secret",
+                                "You have entered an invalid secret key. The key must be in Base32 format.\n"
+                                "Example: JBSWY3DPEHPK3PXP");
+        return;
+    }
+
     QString encShortName;
     uint digits = Totp::DEFAULT_DIGITS;
     uint step = Totp::DEFAULT_STEP;
+    Totp::Algorithm algorithm = Totp::DEFAULT_ALGORITHM;
+    Totp::StorageFormat format = Totp::DEFAULT_FORMAT;
 
     if (m_ui->radioSteam->isChecked()) {
         digits = Totp::STEAM_DIGITS;
         encShortName = Totp::STEAM_SHORTNAME;
     } else if (m_ui->radioCustom->isChecked()) {
+        algorithm = static_cast<Totp::Algorithm>(m_ui->algorithmComboBox->currentData().toInt());
         step = m_ui->stepSpinBox->value();
-        if (m_ui->radio8Digits->isChecked()) {
-            digits = 8;
-        } else if (m_ui->radio7Digits->isChecked()) {
-            digits = 7;
+        digits = m_ui->digitsSpinBox->value();
+    }
+
+    auto settings = m_entry->totpSettings();
+    if (settings) {
+        if (key.isEmpty()) {
+            auto answer = MessageBox::question(this,
+                                               "Confirm Remove TOTP Settings",
+                                               "Are you sure you want to delete TOTP settings for this entry?",
+                                               MessageBox::Delete | MessageBox::Cancel);
+            if (answer == MessageBox::Cancel) {
+                return;
+            }
```

Comment:
`if (answer != MessageBox::Delete) { ... }`

Reply:
No you want to early-out return here because you want to execute the rest of the function to actually delete the settings. Early out brings you back to the dialog which let's you undo or cancel your actions.

Reply:
Gotcha!